### PR TITLE
make Kubernetes use standalone OVS container

### PR DESCRIPTION
**Pull request by unclejack**
This PR makes the netplugin Docker image not use the embedded OVS by default. This makes it possible to run OVS in a separate container.

### CODE REVIEW

File: install/k8s/contiv/contiv_devtest.yaml [hunk 1]
```diff
@@ -338,3 +312,63 @@ spec:
...
+        k8s-app: contiv-ovs
+    spec:
+      hostNetwork: true
+      hostPID: true
+      containers:
+        # Runs netplugin container on each Kubernetes node.  This
+        # container programs network policy and routes on each
+        # host.
+        - name: contiv-ovs
+          image: backjlack/ovs:latest
```
Comment (vhosakot) [inline install/k8s/contiv/contiv_devtest.yaml]: Private image? Is there a plan to use an official image?

File: scripts/netContain/scripts/contivNet.sh [hunk 2]
```diff
@@ -2,6 +2,8 @@
 #Initialize contiv container. Start OVS and netplugin
 set -e
 
+EMBEDDED_OVS=${EMBEDDED_OVS:-"no"}
```
Comment (tiewei) [inline scripts/netContain/scripts/contivNet.sh]: I'd say remove them completely, there's no points to keep ovs running inside the contiv container, it's an upgrading/HA issue

File: install/k8s/contiv/contiv_devtest.yaml [hunk 3]
```diff
@@ -338,3 +312,63 @@ spec:
...
+            path: /etc/openvswitch
+        - name: lib-modules
+          hostPath:
+            path: /lib/modules
+        - name: var-run
+          hostPath:
+            path: /var/run
+        - name: var-contiv
+          hostPath:
+            path: /var/contiv
```
Comment (tiewei) [inline install/k8s/contiv/contiv_devtest.yaml]: This dir seems useless

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by chrisplo** [comment]
build PR

**Comment by dseevr** [comment]
build pr

**Comment by unclejack** [comment]
build PR

**Comment by unclejack** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by unclejack** [comment]
build PR

**Comment by tiewei** [comment]
LGTM !

File: install/k8s/contiv/contiv_devtest.yaml [hunk 4]
```diff
@@ -338,3 +312,57 @@ spec:
...
+        k8s-app: contiv-ovs
+    spec:
+      hostNetwork: true
+      hostPID: true
+      containers:
+        # Runs netplugin container on each Kubernetes node.  This
+        # container programs network policy and routes on each
+        # host.
+        - name: contiv-ovs
+          image: backjlack/ovs
```
Comment (chrisplo) [inline install/k8s/contiv/contiv_devtest.yaml]: need something more transparent and pinned

**Comment by tiewei** [comment]
build PR

**Comment by chrisplo** [comment]
LGTM

**Comment by unclejack** [comment]
I've got some more changes for this.

**Comment by unclejack** [comment]
build PR

File: install/k8s/contiv/contiv_devtest.yaml [hunk 5]
```diff
@@ -338,3 +312,62 @@ spec:
...
+          command: ["/scripts/start-ovsdb-server.sh"]
+          securityContext:
+            privileged: false
+          volumeMounts:
+            - mountPath: /etc/openvswitch
+              name: etc-openvswitch
+              readOnly: false
+            - mountPath: /var/run
+              name: var-run
+              readOnly: false
```
Comment (chrisplo) [inline install/k8s/contiv/contiv_devtest.yaml]: /var/run/openvswitch/db.sock . . that's only used by switch right?  thinking we don't need to declare a mount here, the switch will be on the same filesystem already right?

**Comment by chrisplo** [comment]
thinking, this pod is going to be marked ready before switch is ready to accept traffic, might be room for improvement here using k8s features?

**Comment by tiewei** [comment]
build PR

**Comment by unclejack** [comment]
build PR

**Comment by unclejack** [comment]
build PR

**Comment by unclejack** [comment]
The CI status is green and the PR has been reviewed. I'll merge this now.

**Pull request merged**